src/model.cpp

```diff
@@ -195,7 +217,11 @@ void Model::retireAllVersions(bool isError) {
     for (const auto versionModelInstancePair : modelVersions) {
         SPDLOG_LOGGER_INFO(modelmanager_logger, "Will unload model: {}; version: {} ...", getName(), versionModelInstancePair.first);
         cleanupModelTmpFiles(versionModelInstancePair.second->getModelConfig());
-        versionModelInstancePair.second->unloadModel(true, isError);
+        if (isError) {
+            versionModelInstancePair.second->cleanupFailedLoad();
```

Comment:
Same issue here -> from logs you will get message :

Will unload model
however you will just cleanup if I'm right.